src/wallet/wallet.cpp

```diff
@@ -1327,22 +1327,31 @@ CAmount CWallet::GetCredit(const CTxOut& txout, const isminefilter& filter) cons
 
 bool CWallet::IsChange(const CTxOut& txout) const
 {
-    // TODO: fix handling of 'change' outputs. The assumption is that any
-    // payment to a script that is ours, but is not in the address book
-    // is change. That assumption is likely to break when we implement multisignature
-    // wallets that return change back into a multi-signature-protected address;
-    // a better way of identifying which outputs are 'the send' and which are
-    // 'the change' will need to be implemented (maybe extend CWalletTx to remember
-    // which output, if any, was change).
     if (::IsMine(*this, txout.scriptPubKey))
     {
-        CTxDestination address;
-        if (!ExtractDestination(txout.scriptPubKey, address))
+        CTxDestination dest;
+        if (!ExtractDestination(txout.scriptPubKey, dest))
             return true;
 
         LOCK(cs_wallet);
-        if (!mapAddressBook.count(address))
-            return true;
+        // HD Split assumes keypath of m/.../1(')/k
+        if (CanSupportFeature(FEATURE_HD_SPLIT)) {
+            CKeyID key_id = GetKeyForDestination(*this, dest);
+            if (!key_id.IsNull()) {
+                auto it = mapKeyMetadata.find(key_id);
+                if (it != mapKeyMetadata.end()) {
+                    // Check for path
+                    if (it->second.hdKeypath.find("m/0'/1'/") != std::string::npos) {
```

Comment:
Not for this PR, but we should probably factor out the static key path strings... maybe add a `IsChangeKeypath()`.

Reply:
Yes, I think we should be more systematic about our keypath string handling; would make it easier for people like me with various wallet forks and for Core as we move forward with additional derivation features.